src/structures/ClientApplication.js

```diff
@@ -11,35 +12,64 @@ class ClientApplication extends Application {
   _patch(data) {
     super._patch(data);
 
+    /**
+     * The flags this application has
+     * @type {ApplicationFlags}
+     */
+    this.flags = this.flags ?? new ApplicationFlags(data.flags);
+
     /**
      * The app's cover image
      * @type {?string}
      */
-    this.cover = data.cover_image || null;
+    this.cover = data.cover_image ?? this.cover ?? null;
```

Comment:
This won't update it if the cover is removed

Reply:
There's no way to differenciate between an application being partial or just not having a cover image

Reply:
`Reflect.has(data, 'cover_image') ? data.cover_image : this.cover ?? null`

Reply:
But that would set it to null when receiving a partial application

Reply:
> But that would set it to null when receiving a partial application

No.. it'd set it to this.cover or null... Aka previous value or null if non existent

Reply:
Oh yea my bad, but then there's no difference to the current approach because the property is optional, not nullable ([docs](https://discord.com/developers/docs/topics/oauth2#application-object))

Reply:
There is a difference. If I remove my cover image, d.js will still keep it instead of nulling it out...

Reply:
Same thing as with your suggestion. `data.cover_image` is virtually the same as `Reflect.has(data, 'cover_image')` because again, the property is not nullable

Reply:
Pretty sure applications *can* not have cover images. Reflect.has checks if the property is present, no matter its value. ?? triggers for null or undefined, of which the cover image can be...correct me if I'm wrong tho